scripts/Island.gd

```diff
@@ -0,0 +1,20 @@
+tool
+class_name Island
+extends Spatial
+
+const TREES: Array = [
+	preload("res://scenes/Tree1.tscn"),
+	preload("res://scenes/Tree2.tscn"),
+	preload("res://scenes/Maple.tscn")
+]
+
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	var i: int = 0
+	for child in get_children():
+		if child is Spatial:
+			if child.name.begins_with("tree"):
+				var child_tree = TREES[i % TREES.size()].instance()
+				child.add_child(child_tree)
+				i += 1
```

Comment:
Why do you need i? To keep track of the number of trees?

Reply:
This loop is really simple: we walk through all the tree empty objects, and for each one instantiate a kind of tree.

For simplicity/performance, we just use `i modulo TREES.size()` to choose the tree: yes, i does track the number of trees we've instantiated so far, but really it's about scattering the different tree types. Really we probably want different distributions, so this logic WILL change!